Upgraded Omada Controller and Omada Exporter - Exporter No Longer Runs

Hi,
I've just upgraded my Omada Controller to version 5.14.20.9 and exporter to v0.13.0.
However, I now get the following error when trying to run the exporter:
sudo /bin/omada-exporter --host https://my-omada-server:8043 --username prometheus --password REDACTED --insecure true

panic: runtime error: slice bounds out of range [1:0]

goroutine 1 [running]:
github.com/charlie-haley/omada_exporter/pkg/api.(*Client).getCid(0x40000b2680)
        github.com/charlie-haley/omada_exporter@v0.13.0/pkg/api/auth.go:64 +0x1a4
github.com/charlie-haley/omada_exporter/pkg/api.Configure(0x8dd660)
        github.com/charlie-haley/omada_exporter@v0.13.0/pkg/api/api.go:51 +0x88
github.com/charlie-haley/omada_exporter/cmd.run(0x44b9a0?)
        github.com/charlie-haley/omada_exporter@v0.13.0/cmd/exporter.go:91 +0x19c
github.com/urfave/cli/v2.(*App).RunContext(0x4000082680, {0x5bda68?, 0x4000120020}, {0x4000128000, 0x9, 0x9})
        github.com/urfave/cli/v2@v2.3.0/app.go:322 +0x6ec
github.com/urfave/cli/v2.(*App).Run(...)
        github.com/urfave/cli/v2@v2.3.0/app.go:224
github.com/charlie-haley/omada_exporter/cmd.Run()
        github.com/charlie-haley/omada_exporter@v0.13.0/cmd/exporter.go:60 +0xca4
main.main()
        github.com/charlie-haley/omada_exporter@v0.13.0/main.go:8 +0x1c

Is there a fix for this?

Thanks,
Richie

--- a/pkg/api/auth.go
+++ b/pkg/api/auth.go
@@ -6,7 +6,6 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-	"strings"
 )
 
 func (c *Client) IsLoggedIn() (bool, error) {
@@ -41,27 +40,37 @@ func (c *Client) IsLoggedIn() (bool, error) {
 }
 
 // one of the "quirks" of the omada API - it requires a CID to be part of the path
-// fetching this from the path after redirect seems like the best way
 func (c *Client) getCid() (string, error) {
-	host := c.Config.Host
-	req, err := http.NewRequest("GET", host, nil)
+	url := fmt.Sprintf("%s/api/info", c.Config.Host)
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		return "", err
 	}
 
-	res, err := c.httpClient.Do(req)
+	res, err := c.makeRequest(req)
 	if err != nil {
 		return "", err
 	}
+
 	defer res.Body.Close()
 
-	location := res.Request.URL.Path
+	var infoResponse struct {
+		ErrorCode int    `json:"errorCode"`
+		Msg       string `json:"msg"`
+		Result    struct {
+			OmadaCID string `json:"omadacId"`
+		}
+	}
+	err = json.NewDecoder(res.Body).Decode(&infoResponse)
+	if err != nil {
+		return "", err
+	}
 
-	// remove `/login` path from url
-	location = strings.Replace(location, "/login", "", -1)
+	if infoResponse.Result.OmadaCID == "" {
+		return "", fmt.Errorf("no CID found in response")
+	}
 
-	// trim first rune from string - leading `/`
-	return location[1:], nil
+	return infoResponse.Result.OmadaCID, nil
 }
 
 func (c *Client) Login() error {

